### PR TITLE
[Constraint] Apply new factory registration mechanism

### PR DESCRIPTION
**Pull request by fredroy**
See
- #4429

[ci-depends-on https://github.com/sofa-framework/SofaPython3/pull/462]

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][with-all-tests][force-full-build]

File: Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/StopperLagrangianConstraint.cpp [hunk 1]
```diff
@@ -29,14 +29,13 @@ namespace sofa::component::constraint::lagrangian::model
...
 using namespace sofa::defaulttype;
-using namespace sofa::helper;
 
-int StopperLagrangianConstraintClass = core::RegisterObject("TODO-StopperLagrangianConstraint")
-        .add< StopperLagrangianConstraint<Vec1Types> >()
-
-        ;
+void registerStopperLagrangianConstraint(sofa::core::ObjectFactory* factory)
+{
+    factory->registerObjects(core::ObjectRegistrationData("TODO-StopperLagrangianConstraint")
```
Comment (alxbilger) [inline Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/StopperLagrangianConstraint.cpp]:
```suggestion
    factory->registerObjects(core::ObjectRegistrationData("Lagrangian-based constraint forcing a 1D DoF to be  inside a given range.")
```

File: Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/UnilateralLagrangianConstraint.cpp [hunk 2]
```diff
@@ -28,14 +28,12 @@ namespace sofa::component::constraint::lagrangian::model
...
-
-//TODO(dmarchal) What does this TODO mean ?
-int UnilateralLagrangianConstraintClass = core::RegisterObject("TODO-UnilateralLagrangianConstraint")
-        .add< UnilateralLagrangianConstraint<Vec3Types> >()
-
-        ;
 
+void registerUnilateralLagrangianConstraint(sofa::core::ObjectFactory* factory)
+{
+    factory->registerObjects(core::ObjectRegistrationData("TODO-UnilateralLagrangianConstraint")
```
Comment (alxbilger) [inline Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/UnilateralLagrangianConstraint.cpp]:
```suggestion
    factory->registerObjects(core::ObjectRegistrationData("Lagrangian-based inequality constraint")
```

**Review by alxbilger — CHANGES_REQUESTED** [review]

**Review by epernod — APPROVED** [review]

**Pull request merged**